### ISSUE TITLE
RuntimeError: CUDA error: unspecified launch failure

### ISSUE THREAD

**Issue by JYH9351** [pytorch/pytorch#31702]
## 🐛 Bug

Search datasets...
         Original length: 900
         Offset: 0
         Limit: 900
         Final length: 900
Search datasets...
         Original length: 120
         Offset: 0
         Limit: 120
         Final length: 120
Using CuDNN in the experiment.
Traceback (most recent call last):

  File "<ipython-input-5-cb93f5de020e>", line 1, in <module>
    runfile('C:/Users/Administrator/Desktop/haoxiang_CRN/train.py', wdir='C:/Users/Administrator/Desktop/haoxiang_CRN')

  File "C:\Users\Administrator\Anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 827, in runfile
    execfile(filename, namespace)

  File "C:\Users\Administrator\Anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 110, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/Administrator/Desktop/haoxiang_CRN/train.py", line 103, in <module>
    main(config, resume=resume)

  File "C:/Users/Administrator/Desktop/haoxiang_CRN/train.py", line 93, in main
    validation_dataloader=valid_data_loader

  File "C:\Users\Administrator\Desktop\haoxiang_CRN\trainer\trainer.py", line 27, in __init__
    super(Trainer, self).__init__(config, resume, model, optimizer, loss_function)

  File "C:\Users\Administrator\Desktop\haoxiang_CRN\trainer\base_trainer.py", line 15, in __init__
    self.model = model.to(self.device)

  File "C:\Users\Administrator\Anaconda3\lib\site-packages\torch\nn\modules\module.py", line 432, in to
    return self._apply(convert)

  File "C:\Users\Administrator\Anaconda3\lib\site-packages\torch\nn\modules\module.py", line 208, in _apply
    module._apply(fn)

  File "C:\Users\Administrator\Anaconda3\lib\site-packages\torch\nn\modules\module.py", line 230, in _apply
    param_applied = fn(param)

  File "C:\Users\Administrator\Anaconda3\lib\site-packages\torch\nn\modules\module.py", line 430, in convert
    return t.to(device, dtype if t.is_floating_point() else None, non_blocking)

RuntimeError: CUDA error: unspecified launch failure

## Expected behavior

When I run on a small data set, the above error occurs when the data set becomes larger.

## Environment

Collecting environment information...
PyTorch version: 1.2.0
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: Microsoft Windows 10 Professional
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin\cudnn64_7.dll

Versions of relevant libraries:
[pip] numpy==1.16.5
[pip] numpydoc==0.9.1
[pip] torch==1.2.0
[pip] torchvision==0.4.0
[conda] blas                      1.0                         mkl    defaults
[conda] mkl                       2019.4                      245    defaults
[conda] mkl-service               2.3.0            py37hb782905_0    defaults
[conda] mkl_fft                   1.0.14           py37h14836fe_0    defaults
[conda] mkl_random                1.1.0            py37h675688f_0    defaults
[conda] torch                     1.2.0                    pypi_0    pypi
[conda] torchvision               0.4.0                    pypi_0    pypi

## Additional context

win10[1903]
Graphics card model：RTX2060
anaconda 4.7.12
Graphics driver 441.20
CUDA:cuda_10.0.130_411.31_win10
CUDNN:cudnn-10.0-windows10-x64-v7.6.0.64
Python:3.7.4 

Do you have any good suggestions for this kind of problem?

**Comment by JYH9351** [comment]
Unable to upload due to large amount of data

**Comment by peterjc123** [comment]
Is your GPU memory sufficient? Where does it stop if you `set CUDA_LAUNCH_BLOCKING=1`?

**Comment by JYH9351** [comment]
The same code works on RTX2080Ti，however,do not work on RTX2060.

**Comment by peterjc123** [comment]
Well, RTX 2060 only has 6GB memory, while 2080 Ti has 11 GB memory. You can use "C:\Program Files\NVIDIA Corporation\NVSMI\nvidia-smi.exe" to monitor the usage of the GPU memory. Closing as it doesn't seem to be the problem at our side.

**Comment by JYH9351** [comment]
> 好吧，RTX 2060只有6GB内存，而2080 Ti有11GB内存。您可以使用“ C：\ Program Files \ NVIDIA Corporation \ NVSMI \ nvidia-smi.exe dmon”来监视GPU内存的使用情况。关闭似乎不是我们这边的问题。

我觉得如果是显存问题的话，报错信息应该不是这个cuda 启动问题，或者说有没有可能是版本的匹配问题呢?

**Comment by peterjc123** [comment]
The CUDA operations are asynchronous, so the original stack frame may be lost when cuda synchronization is done, and thus an `unspecified launch error` is raised instead of an OOM error. Since you didn't give any insights about your issue, I can only make guesses according to my experience. The most likely causes for `unspecified launch error` are
1. Out of GPU memory when trying to `cudaAlloc` (It is easy to monitor using `nvidia-smi`.)
2. TDR timeout on Windows https://zhuanlan.zhihu.com/p/38141415 (You can try this first.)
3. Error in your code that you are not realized (It seems not the case since it works well on one of your machines.)
4. Error in our codebase (However, this is very unlikely because we have tests running all the time.)

**Comment by peterjc123** [comment]
> 或者说有没有可能是版本的匹配问题呢?

It is impossible. We target all the cards with CC 3.7 and up. BTW, please translate your post to English so more devs here know what you mean.

**Comment by alexeygolyshev** [comment]
> Well, RTX 2060 only has 6GB memory, while 2080 Ti has 11 GB memory. You can use "C:\Program Files\NVIDIA Corporation\NVSMI\nvidia-smi.exe dmon" to monitor the usage of the GPU memory. Closing as it doesn't seem to be the problem at our side.

https://github.com/pytorch/pytorch/issues/27837

`nvidia-smi.exe dmon`

```
# gpu   pwr gtemp mtemp    sm   mem   enc   dec  mclk  pclk
# Idx     W     C     C     %     %     %     %   MHz   MHz
    0    27    63     -    18     2     0     0  6801   960
    0    27    63     -    16     1     0     0  6801   960
    0    28    63     -    15     1     0     0  6801   960
    0    28    63     -    16     1     0     0  6801   960
    0    28    63     -    15     1     0     0  6801   960
    0    28    63     -    19     3     0     0  6801   960
    0    27    63     -    15     1     0     0  6801   960
    0    27    63     -    14     0     0     0  6801   960
    0    27    63     -    17     2     0     0  6801   960
    0    27    63     -    14     3     0     0  6801   960
    0    27    63     -     0     0     0     0  6801   960
    0     8    62     -     0     0     0     0   405   345
    0     5    61     -     0     0     0     0   405   300
    0     5    61     -     1     0     0     0   405   300
```

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-6-0c2b063a0352> in <module>
     37         cum_loss.append(loss.item())
     38 
---> 39         loss.backward()
     40         optimizer.step()
     41 

C:\Anaconda3\envs\torch13\lib\site-packages\torch\tensor.py in backward(self, gradient, retain_graph, create_graph)
    148                 products. Defaults to ``False``.
    149         """
--> 150         torch.autograd.backward(self, gradient, retain_graph, create_graph)
    151 
    152     def register_hook(self, hook):

C:\Anaconda3\envs\torch13\lib\site-packages\torch\autograd\__init__.py in backward(tensors, grad_tensors, retain_graph, create_graph, grad_variables)
     97     Variable._execution_engine.run_backward(
     98         tensors, grad_tensors, retain_graph, create_graph,
---> 99         allow_unreachable=True)  # allow_unreachable flag
    100 
    101 

RuntimeError: CUDA error: unspecified launch failure
```

**Comment by peterjc123** [comment]
So this is a duplicate of https://github.com/pytorch/pytorch/issues/27837, right? Let's post there and make the answers appear in one thread.

**Comment by pixixiaonaogou** [comment]
Hi, guys, the error of GPU fan should be fixed now. After Tobias, re-seat the GPU, the error still happened. So, I tried to fix it from software. I found this error usually happened with another error : CUDA error: unspecified launch failure and during the training of my Pytorch code. So, I guess it possibly caused by the mismatch problem between CUDA and Pytorch. So, I used the nvcc -V to show the version of ****CUDA compilation tool and it is 10.1 and used the torch.version.cuda to show the default version of Pytorch's CUDA and it is 10.2. Then, I used this command line conda install pytorch==1.7.1 torchvision==0.8.2 torchaudio==0.7.2 cudatoolkit=10.1 -c pytorch to re-install my Pytorch. Finally, everything is ok.
